contracts/DxMgnPool.sol

```diff
@@ -109,7 +113,9 @@ contract DxMgnPool is Ownable {
             dx.deposit(address(depositToken), depositAmount);
         }
 
-        if (lastParticipatedAuctionIndex != 0) {
+        if (lastParticipatedAuctionIndex != 0 &&
+            dx.sellerBalances(buyToken, sellToken, lastParticipatedAuctionIndex, address(this)) > 0 )
```

Comment:
I wonder if it would be better here to use call instead of checking all the requirements. We have found one edge case now, but who says these are all the reason for a revert. Given that this call is really optional, we could make it a call that doesn't throw? What do you think?

You are also missing another invocation of claimSellerFunds in the method that claims the MGN.

Reply:
The disadvantage of the call is that there might be legit scenarios where the claim is supposed to fail. If claiming is called via .call and only the claim is failing, but the whole function call is not throwing, we can never claim the tokens from the auction.